cloudsync_onedrive.py

```diff
@@ -991,131 +847,127 @@ def create(self, path, file_like, metadata=None) -> 'OInfo':
+        upload_url = r["uploadUrl"]
 
-            data = file_like.read(self.upload_block_size)
+        data = file_like.read(self.upload_block_size)
 
-            max_retries_per_block = 10
+        max_retries_per_block = 10
 
-            cbfrom = 0
-            retries = 0
-            while data:
-                clen = len(data)             # fragment content size
-                cbto = cbfrom + clen - 1     # inclusive content byte range
-                cbrange = f"bytes {cbfrom}-{cbto}/{size}"
-                try:
-                    headers = {"Content-Length": clen, "Content-Range": cbrange}
-                    r = self._direct_api("put", url=upload_url, data=data, headers=headers)
-                except (CloudDisconnectedError, CloudTemporaryError) as e:
-                    retries += 1
-                    log.exception("Exception during _upload_large, continuing, range=%s, exception%s: %s", cbrange, retries, type(e))
-                    if retries >= max_retries_per_block:
-                        raise e
-                    continue
+        cbfrom = 0
+        retries = 0
+        while data:
+            clen = len(data)             # fragment content size
+            cbto = cbfrom + clen - 1     # inclusive content byte range
+            cbrange = f"bytes {cbfrom}-{cbto}/{size}"
+            try:
+                headers = {"Content-Length": clen, "Content-Range": cbrange}
+                r = self._direct_api("put", url=upload_url, data=data, headers=headers)
+            except (CloudDisconnectedError, CloudTemporaryError) as e:
+                retries += 1
+                log.exception("Exception during _upload_large, continuing, range=%s, exception%s: %s", cbrange, retries, type(e))
+                if retries >= max_retries_per_block:
+                    raise e
+                continue
 
-                data = file_like.read(self.upload_block_size)
-                cbfrom = cbto + 1
-                retries = 0
-            return r
+            data = file_like.read(self.upload_block_size)
+            cbfrom = cbto + 1
+            retries = 0
+        return r
 
     def download(self, oid, file_like):
-        with self._api() as client:
-            info = self._get_item(client, oid=oid)
-            r = self._direct_api("get", info.api_path + "/content", stream=True)
-            for chunk in r.iter_content(chunk_size=4096):
-                file_like.write(chunk)
-                file_like.flush()
+        api_path = self._api_path(oid=oid) + "/content"
+        r = self._direct_api("get", api_path, stream=True)
+        for chunk in r.iter_content(chunk_size=4096):
+            file_like.write(chunk)
+            file_like.flush()
 
     def rename(self, oid, path):  # pylint: disable=too-many-locals, too-many-branches, too-many-statements
-        with self._api() as client:
```

Comment:
This provided a mutex around all the setup and the actual rename, preventing race conditions

Reply:
same, i put it back